conjure-java-core/src/main/java/com/palantir/conjure/java/types/MethodSpecs.java

```diff
@@ -38,6 +40,8 @@
 
 public final class MethodSpecs {
 
+    private static final ImmutableSet<String> COLLECTION_TYPES =
+            ImmutableSet.of("java.util.List", "java.util.Set", "java.util.Map");
```

Comment:
I wonder if we could pass through the original conjure type information so that we can avoid regressions if we eventually specialize the field types?

Reply:
We could potentially pipe the `List<EnrichedField>` through and use a `DefaultTypeVisitor` to determine if field should be treated as a collection. It might be worth piping something like `List<Pair<EnrichedField, FieldSpec>>` through to ensure proper association between the two.